### PR TITLE
Allow to filter activities by an object type and id

### PR DESCRIPTION
**Pull request by nickvergessen**
Required for the sidebar activity list #358 

- [x] Depends on #362 
- [ ] Depends on https://github.com/owncloud/core/pull/18372

### Todo
- [x] Make the object type and id possible via the IExtension and IConsumer interfaces
- [x] Add test coverage

### CODE REVIEW

**Comment by karlitschek** [comment]
very nice :+1:

**Comment by PVince81** [comment]
@nickvergessen please rebase to include the changes from https://github.com/owncloud/activity/pull/362

**Comment by nickvergessen** [comment]
Tests will fail, they need to be adjusted and the branch for testing needs to be changed.

**Comment by nickvergessen** [comment]
Test should pass in https://github.com/owncloud/activity/pull/366

File: appinfo/database.xml [hunk 1]
```diff
@@ -89,6 +89,19 @@
...
+				<name>object_type</name>
+				<type>text</type>
+				<notnull>false</notnull>
+				<length>255</length>
+			</field>
+			<field>
+				<name>object_id</name>
+				<type>integer</type>
+				<default>0</default>
+				<notnull>true</notnull>
```
Comment (PVince81) [inline appinfo/database.xml]: Can there be (future/third party) activities not linked to any object ?

Comment (PVince81) [inline appinfo/database.xml]: Ah, they'd just have "0" as a value, fine by me.

**Comment by PVince81** [comment]
The cool thing about having the fileid available now (as objectid) is that if we ever implement permalinks https://github.com/owncloud/core/issues/11732 it will be possible to link back to the file from the activity list, regardless where it is located. :smile:

**Comment by karlitschek** [comment]
Indeed.  lot nicer! :-)

**Comment by PVince81** [comment]
@nickvergessen travis no happy ?

**Comment by PVince81** [comment]
ah of course, needs to have the master one merged first?

**Comment by nickvergessen** [comment]
@PVince81 travis is happy here:
https://github.com/owncloud/activity/issues/366

Only additional commit is changing the owncloud branch on travis

**Comment by PVince81** [comment]
:+1:

**Comment by PVince81** [comment]
tested here: https://github.com/owncloud/core/pull/18372#issuecomment-133000001

**Comment by MorrisJobke** [comment]
Tested in https://github.com/owncloud/core/pull/18372#issuecomment-133012261 :+1:

**Comment by MorrisJobke** [comment]
I guess you want to increase https://github.com/owncloud/activity/blob/master/appinfo/info.xml#L23 too ;)

**Pull request merged**